### DISCUSSION TITLE
新手入坑 taro，现在时间是选择 v3 还是 v4 呢？react 兼容 会比 vue 更好吗？

### DISCUSSION THREAD

**Discussion by zanezhangzz** [NervJS/taro#16504] [Q & A]
2月份看，有说 Q2 v4 就要发布了，有知道延后到什么时候的吗？

**Comment by TheKonka** [comment]
个人 推荐 v3 + react

**Comment by hanjin001** [comment]
其实已经发布了 `taro v4` ，`4.0.3` 是 `latest` 的第一个版本。
<img width="1320" alt="image" src="https://github.com/user-attachments/assets/db2635dc-bec6-4b4a-a781-4e86d69d51f4">

只是目前正式的文章还没发布。可以看看 taro 4.0 beta 的文章。[Taro 4.0 Beta 发布：支持开发鸿蒙应用、小程序编译模式、Vite 编译等](https://juejin.cn/post/7330792655125463067)

目前最新版本是 [4.0.5](https://github.com/NervJS/taro/releases)

<img width="721" alt="image" src="https://github.com/user-attachments/assets/f6e926ca-b724-4e5f-b3cc-faf94ed67ae5">


觉得自己有能力解决一些坑（暂时应该没多少坑），可以直接使用最新版本 4.0.5了，以免以后还需要升级。v4 + react + webpack 比较推荐 （`npx @tarojs/cli init project` 初始化可以使用CLI默认模版）。

**Reply by zanezhangzz** [reply]
研究了下，项目模版初始化有问题，文档还不全，还是等专业的前端先踩坑吧

**Reply by noneorone** [reply]
4.x正式版官方还没有正式发文，目前用最新的4.0.5版本默认模版好像报错，在原有的3.6.35基础上直接taro update self 好像过后运行的时候也会报错。

**Reply by hanjin001** [reply]
> 4.x正式版官方还没有正式发文，目前用最新的4.0.5版本默认模版好像报错，在原有的3.6.35基础上直接taro update self 好像过后运行的时候也会报错。

报什么错呢，使用的哪个模板呢~

**Comment by ShaoGongBra** [comment]
选择react会更好，只有react能开发 app和鸿蒙
可以来试试我这个框架 http://duxapp.cn

**Comment by xxxifan** [comment]
希望能提供一个3.x稳定版本的直接支持，我也是新入坑，目前4.x连文档都不全，按照3.x的文档直接装上了4.0

**Comment by oriental1212** [comment]
我在4.0版本安装的时候根本找不到react-native的模板构建脚本一直报错。。。

**Comment by ShaoGongBra** [comment]
> 我在4.0版本安装的时候根本找不到react-native的模板构建脚本一直报错。。。

用duxapp支持app
